text/0000-lifecycle-experimental-api-mode.md

```diff
@@ -0,0 +1,122 @@
+
+New `CNB_PLATFORM_EXPERIMENTAL_MODE` and `CNB_BUILDPACK_EXPERIMENTAL` mode environment variables will control experimental mode with:
+allowed values: `warn`, `error`, `silent`
+default value: `warn`
+
+**When** the `CNB_PLATFROM_API` environment variable is set to an API version in the deprecated platform API, the lifecycle shall:
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE` is unset, **Then** print a warning and continue
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE=warn`, **Then** print a warning and continue
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE=error`, **Then** fail
+ - **If** `CNB_PLATFORM_EXPERIMENTAL_MODE=silent`, **Then** continue w/o warning
+
+**When** the `api` field in a `buildpack.toml` file is set to an API version in the deprecated buildpack API range the lifecycle shall:
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE` is unset, **Then** print a warning and continue
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=warn`, **Then** print a warning and continue
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=error`, **Then** fail
+ - **If** `CNB_BUILDPACK_EXPERIMENTAL_MODE=silent`, **Then** continue w/o warning
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The technical implementation will follow what's outlined in [RFC#49](https://github.com/buildpacks/rfcs/blob/main/text/0049-multi-api-lifecycle-descriptor.md).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+This will add more complexity/overhead that most users will not use.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Do Nothing
+If we choose to do nothing, we run the risk of releasing less tested API/spec releases.
+
+### Not a Different Version Format
+The Experimental API version format could instead follow the same format as deprecated and supported: `<major>-<minor>`. By doing this, users could automatically get upgraded from an experimental to supported/deprecated API version which may be surprising to them.
+
+### Unstable Features
+Similar to [RFC#91](https://github.com/buildpacks/rfcs/pull/91), instead of tying features to a specific release, features can be tied behind flags for testing. There's a potential this could be complicated to do if the features are coupled to other changes going on in the API/spec. The benefit is folks can try out individual features in a more explicit way.
+
+# Prior Art
+[prior-art]: #prior-art
+
+## Rust Unstable Features
+Rust has the concept of [Unstable Features](https://doc.rust-lang.org/beta/unstable-book/the-unstable-book.html) where they have to be explicitly enabled in code. They're all documented in one place.
+
+## Heroku API Stability Attribute
+The Heroku API marks the status of every API endpoint with the [`stability' attribute](https://devcenter.heroku.com/articles/platform-api-reference#stability).
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+How does an experimental version map to the branch?
+Is there a better word than "experimental"?
```

Comment:
it's common to use `X` in place of "experimental if we want it to be shorter. But I don't think `CNB_PLATFORM_X_MODE` would be obvious to most people.

Also could use "preview"